source/frontend/src/features/mapSideBar/acquisition/tabs/agreement/detail/AgreementView.tsx

```diff
@@ -1,148 +1,224 @@
-          </SectionField>
-          {agreement.agreementStatusType?.id ===
-            ApiGen_CodeTypes_AgreementStatusTypes.CANCELLED && (
-            <SectionField labelWidth="5" label="Cancellation reason">
-              {agreement.cancellationNote ?? ''}
-            </SectionField>
-          )}
-          <SectionField labelWidth="5" label="Legal survey plan">
-            {agreement.legalSurveyPlanNum}
-          </SectionField>
-          <SectionField labelWidth="5" label="Agreement type">
-            {agreement.agreementType?.description}
-          </SectionField>
-          <SectionField labelWidth="5" label="Agreement date">
-            {prettyFormatDate(agreement.agreementDate)}
-          </SectionField>
-          {agreement.agreementType?.id === H0074Type && (
-            <SectionField labelWidth="5" label="Commencement date">
-              {prettyFormatDate(agreement.commencementDate)}
-            </SectionField>
-          )}
-          <SectionField labelWidth="5" label="Completion date">
-            {prettyFormatDate(agreement.completionDate)}
-          </SectionField>
-          <SectionField labelWidth="5" label="Termination date">
-            {prettyFormatDate(agreement.terminationDate)}
-          </SectionField>
-          <SectionField labelWidth="5" label="Possession date">
-            {prettyFormatDate(agreement.possessionDate)}
-          </SectionField>
 
-          <StyledSectionSubheader>Financial</StyledSectionSubheader>
-          <SectionField labelWidth="5" label="Purchase price">
-            {formatMoney(agreement.purchasePrice)}
-          </SectionField>
-          <SectionField
-            labelWidth="5"
-            label="Deposit due no later than"
-            tooltip="Generally, if applicable, this is number of days from the execution of the agreement."
-          >
-            {agreement.noLaterThanDays ? (
-              <span>
-                {agreement.noLaterThanDays} <strong>days</strong>
-              </span>
-            ) : (
-              ''
-            )}
-          </SectionField>
-          <SectionField labelWidth="5" label="Deposit amount">
-            {formatMoney(agreement.depositAmount)}
-          </SectionField>
-        </Section>
-      ))}
+      <Section
+        header={
+          <SectionListHeader
+            claims={[Claims.ACQUISITION_EDIT]}
+            title="Agreements"
+            addButtonText="Add Agreement"
+            addButtonIcon={<FaPlus size={'2rem'} />}
+            onAdd={() => {
+              history.push(`${match.url}/add`);
+            }}
+          />
+        }
+      >
+        {agreements.map((agreement, index) => (
+          <StyledAgreementBorder key={`agreement-section-${index}`}>
+            <Section
+              header={
+                <Row>
+                  <Col md={5}>{`Agreement ${++index}`}</Col>
+                  <Col md={7}>
+                    {agreement.agreementType !== null && (
```

Comment:
exists(agreement.agreementType) is safer than just checking for null

Reply:
updated